homework/index.js

```diff
@@ -16,31 +42,86 @@
+    divRepoInfo.innerHTML = '';
+    const table = createAndAppend('table', divRepoInfo);
+    const tbody = createAndAppend('tbody', table);
+
+    const firstRow = addRow(tbody, 'Repository', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repos.html_url,
+      target: '_blank',
+      text: repos.name,
+    });
+
+    addRow(tbody, 'Description', repos.description);
+    addRow(tbody, 'Forks', repos.forks);
+    addRow(tbody, 'Updated', new Date(repos.updated_at).toLocaleString());
+  }
+
+  function renderContributors(contributorRepo, ulContributorInfo) {
+    const url = contributorRepo.contributors_url;
+    fetchJSON(url, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+      ulContributorInfo.innerHTML = '';
+
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', ulContributorInfo, {
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        const listItem = createAndAppend('li', contributorLink, {
+          class: 'contributor-list-item',
+        });
+        createAndAppend('img', listItem, {
+          src: contributor.avatar_url,
+          alt: `${contributor.login} photo`,
+        });
+        createAndAppend('h1', listItem, { text: contributor.login });
+        createAndAppend('p', listItem, { text: contributor.contributions });
+      });
+    });
+  }
+
+  function createOptionElements(repositories, select) {
+    repositories.forEach((repo, index) => {
+      createAndAppend('option', select, { text: repo.name, value: index });
     });
-    return elem;
   }
 
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    const infoSection = createAndAppend('section', root, { class: 'wrapper' });
+    const headerWrapper = createAndAppend('div', header, { class: 'header-wrapper' });
+    createAndAppend('h1', headerWrapper, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', headerWrapper);
+    const repoInfo = createAndAppend('div', infoSection, { class: 'flex-container' });
+    const contributorInfo = createAndAppend('ul', infoSection, {
+      class: 'flex-container',
+    });
+
     fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        renderError(err);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      createOptionElements(repositories, select);
+      const repo = repositories[select.value];
+      renderRepositories(repo, repoInfo);
+      renderContributors(repo, contributorInfo);
+
+      select.addEventListener('change', () => {
+        // const repo = repositories[select.value];
```

Comment:
Thank you again your kind review. i worked different files. I think I tested another file. Because of that reason I skipped original file. I will be more careful than that. Sorry for that.